File: swift/1837 - Sum of Digits in Base K.swift
/* Method 1: */
class Solution {
    func sumBase(_ n: Int, _ k: Int) -> Int {
        var n = n
        var result = [Int]()

        while n > 0 {
            result.append(n % k)
            n /= k
        }

        return result.reduce(0,+)
     
    }
}

// Method 2:
class Solution {
    func sumBase(_ n: Int, _ k: Int) -> Int {
        var n = n
        var result = ""
        
        while n > 0 {
            let r = n % k
            result.append(String(r))
            n /= k
        }
        
        var newBaseNum = Int(String(result.reversed()))!
        
        var sum = 0
        
        while newBaseNum > 0 {
            sum += newBaseNum % 10
            newBaseNum /= 10
        }
        
        return sum
        
    }
}

File: python/1837 - Sum of Digits in Base K.py
# Method 1:
class Solution:
    import functools

    def sumBase(self, n: int, k: int) -> int:
        digits = []
     
        while n > 0:
            digits.append(n%k)
            n //= k
        
        return functools.reduce(operator.add, digits)

# Method 2:      
class Solution:
    def sumBase(self, n: int, k: int) -> int:
        sum = 0
     
        while n > 0:
            sum += (n%k)
            n //= k
        
        return sum
      
      
